src/main/java/carpet/script/annotation/OutputConverter.java

```diff
@@ -0,0 +1,129 @@
+package carpet.script.annotation;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.ClassUtils;
+
+import carpet.script.LazyValue;
+import carpet.script.value.EntityValue;
+import carpet.script.value.FormattedTextValue;
+import carpet.script.value.NBTSerializableValue;
+import carpet.script.value.NumericValue;
+import carpet.script.value.StringValue;
+import carpet.script.value.Value;
+import carpet.script.value.ValueConversions;
+import net.minecraft.entity.Entity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.Tag;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.dynamic.GlobalPos;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+
+/**
+ * <p>A converter from a given {@link Object} of type T into a {@link LazyValue}, used in order to convert the outputs of methods into usable Scarpet
+ * values.</p>
+ *
+ * @see #register(Class, Function)
+ * @see #registerToValue(Class, Function)
+ * @param <T> The type to convert from into a {@link LazyValue}
+ */
+public final class OutputConverter<T>
+{
+    private static final Map<Class<?>, OutputConverter<?>> byResult = new HashMap<>();
+    private static final OutputConverter<Value> VALUE = new OutputConverter<>(v -> (c, t) -> v);
+    static
+    {
+        register(LazyValue.class, Function.identity()); // Primitives are handled. Things are boxed in the process anyway, therefore
```

Comment:
This functionality could be merged with ValueConversions::guess since they share similar functionality, just the guess requires world handler to convert some runtime types, but a map based converter is better than a series of if-elses.

Reply:
Hm... Problem is `ValueConversions#guess` uses `instanceof`, which is more flexible than the class directly like here (currently thinking of having `#guess` call `Object#getClass` and search that way in the map, which wouldn't work for interfaces or subclasses).

The reason it works here is because the class used is the declared one (in the method's return type), so even if a subclass or implementation is given it will still search for the former. Can't think of any other way to do it (other than passing the class as another parameter maybe, which I'm not sure if it's worth it).